### PR TITLE
Dev/document jio 3.6.4

### PR DESCRIPTION
**Pull request by lav-mudgal-guavus**

### CODE REVIEW

File: core-plugins/docs/File-batchsink.md [hunk 1]
```diff
@@ -3,13 +3,16 @@
```
Comment (deepak-maheshwari-ext) [inline core-plugins/docs/File-batchsink.md:3]: Please provide a enduser use case as well.

File: core-plugins/docs/File-batchsink.md [hunk 2]
```diff
@@ -3,13 +3,16 @@
 
 Description
 -----------
-Writes to a filesystem in various formats.
+File Sink is a Guavus Enterprise Accelerator that writes to a filesystem in various formats.
```
Comment (deepak-maheshwari-ext) [inline core-plugins/docs/File-batchsink.md]: change : writes to a filesystem -> writes to hdfs

File: transform-plugins/docs/Compressor-transform.md [hunk 3]
```diff
@@ -3,26 +3,27 @@
 
 Description
 -----------
```
Comment (deepak-maheshwari-ext) [inline transform-plugins/docs/Compressor-transform.md:5]: please provide end user use case here as well . 
Like : apply any source -> compressor transform -> then any sink  (convert this into statements.)

File: transform-plugins/docs/Compressor-transform.md [hunk 4]
```diff
@@ -3,26 +3,27 @@
...
+The accelerator supports SNAPPY, ZIP, and GZIP types of compression.
 
 
 Configuration
 -------------
-**compressor:** List of key value pairs. Key represents the input field that needs to be compressed and Value represents the compression algorithm to use.
+You need to specify the following configuration for the accelerator:
 
-**Output Schema:** Output schema will be same as input schema except fields in output schema that are compressed will have the same field name 
-but they will be of type *Non-nullable* ``BYTES``.
```
Comment (deepak-maheshwari-ext) [inline transform-plugins/docs/Compressor-transform.md:15]: Add getSchema property here as well. which is responsible to get schema on the basis of compressor field and apply that schema in output schema  section . User do not need to change compressor fields manually to bytes in out schema .

File: transform-plugins/docs/Decoder-transform.md [hunk 5]
```diff
@@ -3,20 +3,22 @@
 
 Description
 -----------
```
Comment (deepak-maheshwari-ext) [inline transform-plugins/docs/Decoder-transform.md:5]: add end user use case here a swell as i described in compressor .

File: transform-plugins/docs/Encoder-transform.md [hunk 6]
```diff
@@ -3,21 +3,20 @@
 
 Description
 -----------
```
Comment (deepak-maheshwari-ext) [inline transform-plugins/docs/Encoder-transform.md:5]: add end user use case here as well .

**Review by deepak-maheshwari-ext — CHANGES_REQUESTED** [review]

File: core-plugins/docs/File-batchsink.md (hunk 1 above)
Comment (rohit-raina-guavus) [reply core-plugins/docs/File-batchsink.md:3]: It wasn't in the original file, it's better that the use case  comes from you. I'll edit it language wise if needed

Comment (deepak-maheshwari-ext) [reply core-plugins/docs/File-batchsink.md:3]: so use case should be like this . 
apply any any source plugin(es,file,kafka, etc) -> file sink .

Comment (rohit-raina-guavus) [reply core-plugins/docs/File-batchsink.md:3]: This does not seem to be going with the first line in this document. Please suggest a complete use case

Comment (deepak-maheshwari-ext) [reply core-plugins/docs/File-batchsink.md:3]: Consider a scenario wherein you need to write a file to an HDFS in batch. For example, you may want to periodically dump any RDD data to HDFS in the formate of files(csv,tsv,json etc.) .

**Review by deepak-maheshwari-ext — CHANGES_REQUESTED** [review]

File: transform-plugins/docs/NullFieldSplitter-splittertransform.md [hunk 7]
```diff
@@ -3,62 +3,63 @@
...
+The records containing fields with a null value are sent to the ``null`` port while records with non-null value fields are sent to the ``nonnull`` port.
+
+Use Case
+-----------
+
+Consider a scenario wherein you have a large number of records and you want to separate records with null values from the ones with non-null values. This can be done by configuring the accelerator as explained in the following section.
 
 Properties
 ----------
-**field:** Specifies which field should be checked for null values. (Macro-enabled)
```
Comment (deepak-maheshwari-ext) [inline transform-plugins/docs/NullFieldSplitter-splittertransform.md:12]: add "generate schema" button . which is responsible to genrate output schema on the basis of configuration .

**Review by deepak-maheshwari-ext — CHANGES_REQUESTED** [review]

File: transform-plugins/docs/Encoder-transform.md [hunk 8]
```diff
@@ -3,21 +3,26 @@
...
+**schema:** Specifies the output schema. The fields that are encoded will have the same field name, but they will be of type ``BYTES`` or ``STRING``.
 
 #### Note
-- *Input fields that need to encoded must be of type `String` or `Bytes` and non-nullable*
+- *Input fields that need to be encoded must be of type `String` or `Bytes` and non-nullable*
+
+
+Use Case
+--------
+Consider a scenario wherein you want to encode some fields of your dataset for efficient storage and data transfer. For instance, you apply any source accelerator to take the input as rdd and apply the Encoder accelerator to encode the desired fields, then sink the transformed output using any sink accelerator.
```
Comment (deepak-maheshwari-ext) [inline transform-plugins/docs/Encoder-transform.md]: efficient storage and data transfer -> for securing data field from external user.

**Review by deepak-maheshwari-ext — CHANGES_REQUESTED** [review]

**Review by deepak-maheshwari-ext — APPROVED** [review]

**Pull request merged**